src/main/java/io/vertx/kafka/client/common/tracing/ConsumerTracer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.vertx.kafka.client.common.tracing;
+
+import io.vertx.codegen.annotations.Nullable;
+import io.vertx.core.Context;
+import io.vertx.core.spi.tracing.TagExtractor;
+import io.vertx.core.spi.tracing.VertxTracer;
+import io.vertx.core.tracing.TracingPolicy;
+import io.vertx.kafka.client.common.KafkaClientOptions;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.utils.Utils;
+
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.StreamSupport;
+
+/**
+ * Tracer for Kafka consumer, wrapping the generic tracer.
+ */
+public class ConsumerTracer<S> {
+  private final VertxTracer<S, Void> tracer;
+  private final String address;
+  private final String hostname;
+  private final String port;
+  private final TracingPolicy policy;
+
+  public static <S> @Nullable ConsumerTracer create(VertxTracer tracer, KafkaClientOptions opts) {
```

Comment:
no need for `Nullable` here as it is not processed by code generator

Reply:
that was essentially informative, but I'm gonna add more javadoc instead

Reply:
done